### ISSUE TITLE
Building with Bitcoin 0.12.1 fails

### ISSUE THREAD

**Issue by smouldering** [shadow/shadow-plugin-bitcoin#4]
In the _shadow-plugin-bitcoin_ directory after using the command _make -j1_ I get an error stating:-

> make[2]: **\* No rule to make target `../bitcoin/src/verison.cpp', needed by `src/version.cpp.bc'. Stop.

Maybe this _verison.cpp_ doesn't exist in Bitcoin 0.9.2 netmine edition and needs to be added to some cmake config file?

**Comment by smouldering** [comment]
Seems _version.cpp_ doesn't exist in later versions. Any pointers to what would need to be modified in the builder I'd be grateful.

**Comment by robgjansen** [comment]
If you just want to remove that file from the build, you could comment out this line:
https://github.com/shadow/shadow-plugin-bitcoin/blob/master/src/CMakeLists.txt#L143

I have no idea if that will cause other problems though.

**Comment by smouldering** [comment]
Ah I see. Looking at the code I might have to add a lot of sources too, ill comment back on my findings.

**Comment by robgjansen** [comment]
It may be easier to grab all sources in the source directories, and then remove the ones you don't want. We do that for the Tor plugin:
https://github.com/shadow/shadow-plugin-tor/blob/master/src/tor/CMakeLists.txt#L136

**Comment by smouldering** [comment]
Yeah that does sound easier, my assumption is to write _aux_source_directory(${BITCOIN_ROOT_DIR}/src bitcoin_sources)_. I'll give that a try.

EDIT: Just seen it's already in there just commented out: https://github.com/shadow/shadow-plugin-bitcoin/blob/master/src/CMakeLists.txt#L93

**Comment by smouldering** [comment]
It does pick up the sources and starts to build them but, it cannot build _key.cpp_ as 'secp256k1.h' doesn't exist. So, like the Bitcoin make file, the dependencies need building first and somehow linked to building the rest of the Bitcoin source.

EDIT: My guess is each new source folder will be added tt the CMakeLists.txt file in the same way leveldb was originally.
